lib/puppet/provider/mysql_user/mysql.rb

```diff
@@ -224,7 +240,11 @@ def plugin=(string)
       end
     elsif newer_than('mysql' => '5.7.6', 'percona' => '5.7.6', 'mariadb' => '10.2.0')
       sql = "ALTER USER #{merged_name} IDENTIFIED WITH '#{string}'"
-      sql += " AS '#{@resource[:password_hash]}'" if string == 'mysql_native_password'
+      if string == 'mysql_native_password'
+        sql += " AS '#{@resource[:password_hash]}'"
+      elsif string == 'caching_sha2_password'
+        sql += " AS X'#{@resource[:password_hash][2..-1]}'"
+      end
```

Comment:
I think the ruby world prefers this, but I'm no expert:

```suggestion
      sql+= if string == 'mysql_native_password'
        " AS '#{@resource[:password_hash]}'"
      elsif string == 'caching_sha2_password'
        " AS X'#{@resource[:password_hash][2..-1]}'"
      end
```

Reply:
I had problems with the linting, I think it was regarding this but I am not sure anymore